pkg/controller/deployment/deployment_controller.go

```diff
@@ -1294,3 +1413,27 @@ func (dc *DeploymentController) rollbackToTemplate(deployment *extensions.Deploy
 	d, err = dc.updateDeploymentAndClearRollbackTo(deployment)
 	return
 }
+
+// isScalingEvent checks whether the provided deployment has been updated with a scaling event
+// by looking at the desired-replicas annotation in the active replica sets of the deployment.
+func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) bool {
+	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+	if err != nil {
+		return false
+	}
+	allRSs := append(oldRSs, newRS)
+	for _, rs := range controller.FilterActiveReplicaSets(allRSs) {
+		if desiredString, ok := rs.Annotations[deploymentutil.DesiredReplicasAnnotation]; ok {
+			desired, err := strconv.Atoi(desiredString)
+			if err != nil {
+				glog.V(3).Infof("Cannot convert the value %q with annotation key %q for the replica set %q",
```

Comment:
Ok, Warning is ok but eventually I believe the type of the log doesn't matter if it is already observable for the sole reason that it is not actionable ie. you cannot have another client act based on it.